feat(components/manifest): separate class properties and methods; validate docsIds; add package information; fix output for property accessors; add repo URLs to each type

Blackbaud-SteveBrush opened the pull request.

blackbaud-sky-build-user commented:
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/3196/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples-playground](https://blackbaud.github.io/skyux-pr-preview/3196/code-examples-playground/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/3196/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/3196/playground/)

Blackbaud-SteveBrush commented on scripts/__snapshots__/deprecations.json:
```diff
@@ -325,12 +293,10 @@
   "@skyux/modals:SkyModalHostService:openModalCount",
   "@skyux/modals:SkyModalInstance:helpOpened",
   "@skyux/modals:SkyModalInstance:openHelp",
-  "@skyux/modals:SkyModalInstance:openHelp",
```
☝️ Removing duplicate entries.

Blackbaud-SteveBrush committed:
feat(components/manifest): separate class properties and methods

separate class properties and methods; validate docsIds; add package information; fix output for
property accessors; add repo URLs to each type

Blackbaud-SteveBrush committed:
Merge branch 'main' into manifest-fixes

Blackbaud-SteveBrush committed:
Merge branch 'main' into manifest-fixes
